### ISSUE TITLE
Provision dashboards from file in addition to Grafana Dashboards

### ISSUE THREAD

**Issue by aman0019** [cloudalchemy/ansible-grafana#54]
Currently dashboards can only be provisioned through their dashboard_id and downloaded from the Grafana Dashboards . It would be nice (and pretty simple to implement) if the dashboards can be supplied as a json file to the playbook

**Comment by paulfantom** [comment]
You're right. It should be implemented as a part of #30 but I don't have time right now to do this. Feel free to implement it :smile:  

Grafana even mentions this in their docs: http://docs.grafana.org/administration/provisioning/#dashboards

**Comment by paulfantom** [comment]
That's right.

**Comment by aman0019** [comment]
Yes; also if I understood well if we move towards the YML file provisioning we would need to delay restarting Grafana until after the dashboard tasks are completed since as opposed to the API; YML provisioning files are only picked up at startup. Is that your understanding as well?

**Comment by aman0019** [comment]
> I would like to limit restarts as much as I can.

makes sense

Apologies for asking twenty questions; just want to make sure I get this right. The API does support adding dashboards to folders...what are the benefits to be gained by moving to the 'provisioning by file' method

**Comment by paulfantom** [comment]
For example, right now we don't have any method to copy custom dashboards and we rely only on grafana.net. This way we can easily copy dashboards and as a bonus we could use ansible internals for detecting changes.

**Comment by aman0019** [comment]
Agreed that detecting changes for dashboards provisioned through files is the most obvious benefit. As Grafana's documentation states; the other benefit is that one can push changes whilst Grafana is not running and these would be picked up on startup. However, whilst dashboard JSON files can be added/changed at runtime; the config file that defines `dashboard providers` would only be processed at startup and as such this needs to be provisioned before Grafana is restarted. Do you agree that we should provision a YAML config file that defines the ansible file dashboard provider before startup? and then we can provision dashboards using files in the `dashboard.yml` play.

**Comment by paulfantom** [comment]
Basically this is the same to what I had in mind. However we don't have to do anything before startup as we could apply handler task to restart grafana when `dashboard providers` are changed. This way we don't have to change order of included files in ansible role and still have `dashboards.yml` after task which starts grafana.

It would be awesome if grafana could react on SIGHUP to reload its config, this would allow easier provisioning without interrupts.

**Comment by guilhermesteinmuller** [comment]
Hey there! Pretty good feature would be. 

We switched our grafana repo to cloudalchemy recently. In [ours](https://github.com/flaviodsr/grafana-ansible/blob/796eb972d6712143a0430bc7b6a7aec2fbd9d231/tasks/setup_monasca.yml#L118), we used to push the dashboards locally. 

So, I'd appreciate if you guys could take a look and if thinks is interesting, I can find some time to implement like we used to do :)
